File: epi_judge_cpp/buy_and_sell_stock.cc
#include <vector>
#include "test_framework/generic_test.h"
using std::vector;

double BuyAndSellStockOnce(const vector<double>& prices) {
	double maxProfit = 0.0, minPrice = INT_MAX;
	for (auto const& price: prices) {
		minPrice = std::min(price, minPrice);
		maxProfit= std::max(maxProfit, price - minPrice);
	}
	return maxProfit;
}

int main(int argc, char* argv[]) {
  std::vector<std::string> args{argv + 1, argv + argc};
  std::vector<std::string> param_names{"prices"};
  return GenericTestMain(args, "buy_and_sell_stock.cc",
                         "buy_and_sell_stock.tsv", &BuyAndSellStockOnce,
                         DefaultComparator{}, param_names);
}

File: epi_judge_python/buy_and_sell_stock.py
from test_framework import generic_test


def buy_and_sell_stock_once(prices):
	max_profit = 0.0
	min_price = float('inf')
	for price in prices:
		min_price = min(min_price, price)
		max_profit= max(max_profit, price - min_price)
	return max_profit


if __name__ == '__main__':
    exit(
        generic_test.generic_test_main("buy_and_sell_stock.py",
                                       "buy_and_sell_stock.tsv",
                                       buy_and_sell_stock_once))
